packages/@simple-dom/serializer/src/index.ts

```diff
@@ -83,14 +106,21 @@ export default class HTMLSerializer {
   }
 
   public text(text: SerializableNode) {
-    return this.escapeText(text.nodeValue!);
+    const escapedText = this.escapeText(text.nodeValue!);
+    if (this.opts.trimWhitespace) {
+      return escapedText[this.opts.trimWhitespace]();
+    }
+    return escapedText;
   }
 
   public rawHTMLSection(text: SerializableNode): string {
     return text.nodeValue!;
   }
 
   public comment(comment: SerializableNode) {
+    if (this.opts.stripComments) {
```

Comment:
FWIW, this option being "all or nothing" seems not ideal. glimmer-vm's rehydration system relies on _some_ comment nodes (specially formatted ones):

For example:

https://github.com/glimmerjs/glimmer-vm/blob/1b5b24e7e39937864bcfd2a764f5a0c247fce3e9/packages/%40glimmer/node/lib/serialize-builder.ts#L34

I'm not 100% sure that we can do something better about it, but we should definitely consider carefully

Reply:
Yeah, it was these reasons (and a few more along the same vein) that led me to make the change to allow custom serializers in glimmer.js instead (an rb I think you also looked at this morning) I plan on closing this PR in favor of that approach.